### PR TITLE
[FEATURE] Add replyTo field for class Tx_Oelib_Mail

### PR DESCRIPTION
**Pull request by pascal20997**
Add the possibility to define a reply address for a mail.

This property will then be added to the Swift Message.

### CODE REVIEW

File: Classes/Mail.php [hunk 1]
```diff
@@ -46,7 +51,7 @@ class Tx_Oelib_Mail extends \Tx_Oelib_Object
      */
     public function __destruct()
     {
-        unset($this->data, $this->sender, $this->recipients, $this->attachments);
+        unset($this->data, $this->sender, $this->replyTo, $this->recipients, $this->attachments);
     }
```
Comment (oliverklee) [inline Classes/Mail.php]: I think we don't need all these unsets anymore as the PHP garbage collector is in far better shape now that is was 10 years ago. So please let's completely get rid the of the `__destruct__` method in this class altogether.

File: Classes/Mail.php [hunk 2]
```diff
@@ -110,6 +115,37 @@ public function hasSender()
         return is_object($this->sender);
     }
 
+    /**
+     * Returns ReplyTo
```
Comment (oliverklee) [inline Classes/Mail.php]: I think the comments are not needed as the names already are "speaking" enough. Ditto for the other new methods.

File: Classes/Mail.php [hunk 3]
```diff
@@ -110,6 +115,37 @@ public function hasSender()
...
+     *
+     * @return Tx_Oelib_Interface_MailRole|null the reply mail role of the e-mail, will be NULL if it has not been set
```
Comment (oliverklee) [inline Classes/Mail.php]: Please always use a trailing backslash for FQCNs. Ditto for the other new methods.

And I think the explanation for the return value is also not needed anymore.

File: Classes/Mail.php [hunk 4]
```diff
@@ -110,6 +115,37 @@ public function hasSender()
...
+    public function getReplyTo()
+    {
+        return $this->replyTo;
+    }
+
+    /**
+     * Sets ReplyTo
+     *
+     * @param Tx_Oelib_Interface_MailRole $replyTo
+     * @return void
```
Comment (oliverklee) [inline Classes/Mail.php]: Please add a blank line between the param and the return line.

File: Classes/Mail.php [hunk 5]
```diff
@@ -110,6 +115,37 @@ public function hasSender()
...
+     */
+    public function setReplyTo(Tx_Oelib_Interface_MailRole $replyTo)
+    {
+        $this->replyTo = $replyTo;
+    }
+
+    /**
+     * Returns whether the e-mail has a replyTo object.
```
Comment (oliverklee) [inline Classes/Mail.php]: better: "has a reply-to address".

File: Classes/Mail.php [hunk 6]
```diff
@@ -110,6 +115,37 @@ public function hasSender()
...
+     *
+     * @return bool TRUE if the e-mail has a replyTo object, FALSE otherwise
```
Comment (oliverklee) [inline Classes/Mail.php]: This comments also is not needed anymore, I think.

File: Classes/Mail.php [hunk 7]
```diff
@@ -110,6 +115,37 @@ public function hasSender()
...
+     */
+    public function hasReplyTo()
+    {
+        return is_object($this->replyTo);
```
Comment (oliverklee) [inline Classes/Mail.php]: Better: `!== null` (as this will make the negative case more explicit)

File: Tests/Unit/MailTest.php [hunk 8]
```diff
@@ -78,6 +78,61 @@ public function hasSenderWithSenderReturnsTrue()
...
+            'John Doe',
+            'foo@bar.com'
+        );
+
+        $this->subject->setReplyTo($sender);
+
+        self::assertTrue(
+            $this->subject->hasReplyTo()
+        );
+    }
```
Comment (oliverklee) [inline Tests/Unit/MailTest.php]: Beautiful tests! :heart:

File: ext_emconf.php [hunk 9]
```diff
@@ -22,7 +22,7 @@
     'CGLcompliance' => '',
     'CGLcompliance_note' => '',
     'author_company' => 'oliverklee.de',
-    'version' => '2.0.1',
+    'version' => '2.1.0',
```
Comment (oliverklee) [inline ext_emconf.php]: Please don't change the version number in this PR. (I will do this right before the next release).

Please do add a changelog line, though.

File: Classes/Mail.php (hunk 1 above)
Comment (pascal20997) [reply Classes/Mail.php]: That´s a great idea :)

File: Tests/Unit/MailTest.php (hunk 8 above)
Comment (pascal20997) [reply Tests/Unit/MailTest.php]: :grinning:

**Comment by oliverklee** [comment]
@pascal20997 After I've merged this, will you be able to use the current oelib master, or do you need a 2.1.0 release very soon? (I'd like to fix the build timeouts before the next release.)

**Comment by pascal20997** [comment]
@oliverklee I´ll use the master until you release the new version that´s no problem, thanks.

**Pull request merged**